### PR TITLE
[BugFix] support structType matchesType() used in function argument type matches

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/StructTypeTest.java [hunk 1]
```diff
@@ -41,4 +49,58 @@ public void testStructMultiSelect() throws Exception {
...
+        // matched
+        StructType mc1 = new StructType(Lists.newArrayList(
+                new StructField("c1", ScalarType.createType(PrimitiveType.INT)),
+                new StructField("cc1", ScalarType.createDefaultExternalTableString())
+        ));
+        StructType matched = new StructType(Lists.newArrayList(
+                new StructField("struct_test", ScalarType.createType(PrimitiveType.INT)),
+                new StructField("c1", mc1)
+        ));
+        Assert.assertTrue(root.matchesType(matched));
```
Comment (Smith-Cruise) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/StructTypeTest.java:104]: Test different subfield order in struct.

**Review by Smith-Cruise — APPROVED** [review]

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|14|14|100.00%|[]|

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**